src/Contracts/SearchQuery.php

```diff
@@ -26,6 +26,35 @@ class SearchQuery
     private ?int $hitsPerPage;
     private ?int $page;
 
+    // @phpstan-ignore-next-line
+    public function __construct(
+        string $indexUid = null,
+        string $q = null,
+        array $filter = null,
+        array $attributesToRetrieve = null,
+        array $attributesToCrop = null,
+        int $cropLength = null,
+        array $attributesToHighlight = null,
+        string $cropMarker = null,
+        string $highlightPreTag = null,
+        string $highlightPostTag = null,
+        array $facets = null,
+        bool $showMatchesPosition = null,
+        array $sort = null,
+        string $matchingStrategy = null,
+        int $offset = null,
+        int $limit = null,
+        int $hitsPerPage = null,
+        int $page = null,
+    ) {
+        foreach (get_defined_vars() as $name => $value) {
```

Comment:
Set the vars manually then the phpstan won't complain and in future it could be refactored to CPP